cmd/kube-proxy/app/server.go

```diff
@@ -1046,7 +1046,8 @@ func getNodeIPs(logger klog.Logger, client clientset.Interface, name string) []n
 	}
 
 	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
-		node, err := client.CoreV1().Nodes().Get(context.TODO(), name, metav1.GetOptions{})
+		// Improve the efficiency by setting ResourceVersion "0", which means "serve the current version held by the API server cache".
```

Comment:
Please update the comment that the returned version of the node may be stale - it's the most important part - the optimization is not for free.

Reply:
BTW - given it's a one-off thing on initialization, does it matter much?

Reply:
hmm, we had issues with stale data during initialization , it may be better to let this as is https://github.com/kubernetes/kubernetes/pull/118499 and not open a new possible problem

Reply:
> BTW - given it's a one-off thing on initialization, does it matter much?

Yes, it does. In the case where the kube-proxy starts in parallel to kubelet, it continuously queries for node object and hits ETCD, which is observed having >100 ms latency compared to cache query < 1ms when apiserver under stress.

Reply:
What's the user story where cutting node startup by 100ms helps out enough to make this worthwhile?

Reply:
Not only the latency, but also the performance impact to apiserver. Spawning a large number of nodes with non-cache queries to kube-apiserver would result in CPU and memory usage surge, which instablizes whole cluster.

Reply:
on the contrary, if there is stale data the kube-proxy will panic and has to restart, I don' think this improvement   is worth the risk